File: Source/be/ChangelogSaas.Domain/Plans/PlanLimits.cs
using ChangelogSaas.Domain.Enums;

namespace ChangelogSaas.Domain.Plans
{
    public static class PlanLimits
    {
        public static int MaxProjects(SubscriptionPlan plan, bool inTrial = false) => (inTrial || plan == SubscriptionPlan.Pro || plan == SubscriptionPlan.Team) ? (plan == SubscriptionPlan.Team ? int.MaxValue : 3) : 1;

        public static int MaxEntries(SubscriptionPlan plan, bool inTrial = false) =>
            inTrial || plan is SubscriptionPlan.Pro or SubscriptionPlan.Team ? int.MaxValue : 5;

        public static int MaxSubscribers(SubscriptionPlan plan, bool inTrial = false) => plan switch
        {
            SubscriptionPlan.Team => int.MaxValue,
            SubscriptionPlan.Pro => 2000,
            _ => inTrial ? 2000 : 100
        };

        public static bool CanUseCustomDomain(SubscriptionPlan plan, bool inTrial = false) =>
            inTrial || plan is SubscriptionPlan.Pro or SubscriptionPlan.Team;

        public static bool CanRemoveBranding(SubscriptionPlan plan, bool inTrial = false) =>
            inTrial || plan is SubscriptionPlan.Pro or SubscriptionPlan.Team;

        public static bool CanUseApi(SubscriptionPlan plan, bool inTrial = false) =>
            inTrial || plan is SubscriptionPlan.Team;
    }
}

File: Source/fe/src/app/core/utils/plan-limits.ts
import { SubscriptionPlan } from '../services/billing.service';

export const PlanLimits = {
  maxProjects: (plan: SubscriptionPlan) => (plan === 'Team' ? Infinity : plan === 'Pro' ? 3 : 1),
  maxEntries: (plan: SubscriptionPlan) => (plan === 'Free' ? 5 : Infinity),
  maxSubscribers: (plan: SubscriptionPlan) => (plan === 'Team' ? Infinity : plan === 'Pro' ? 2000 : 100),
};
